crates/bitfinity-block-validator/src/lib.rs

```diff
@@ -134,11 +143,62 @@ where
         )
     }
 
-    /// Calculates POW hash based on the given trie state.
+    /// Calculates POW hash
     fn calculate_pow_hash(
         &self,
-        updated_state: reth_trie::HashedPostState,
-    ) -> did::hash::Hash<alloy_primitives::FixedBytes<32>> {
-        todo!()
+        block: &Block,
+        execution_result: ExecutionOutcome,
+    ) -> eyre::Result<did::hash::Hash<alloy_primitives::FixedBytes<32>>> {
+        let state = execution_result.bundle;
+        let state_provider = self.provider_factory.latest().expect("no latest provider");
+        let cache = StateCacheDb::new(StateProviderDatabase::new(StateProviderTraitObjWrapper(
+            &state_provider,
+        )));
+
+        let mut state = StateBuilder::new()
+            .with_database_ref(&cache)
+            .with_bundle_prestate(state)
+            .with_bundle_update()
+            .build();
+
+        let chain_spec = self.provider_factory.chain_spec();
+        let evm_config = EthEvmConfig::new(chain_spec);
+
+        let EvmEnv { mut cfg_env_with_handler_cfg, block_env } =
+            evm_config.cfg_and_block_env(&block.header);
+
+        cfg_env_with_handler_cfg.cfg_env.disable_balance_check = true;
+
+        // Simple transaction
+        let tx = TxEnv {
+            caller: Address::from_slice(&[1]),
+            gas_limit: 21000,
+            gas_price: U256::from(1),
+            transact_to: alloy_primitives::TxKind::Call(Address::from_slice(&[2; 20])),
+            value: U256::from(1),
+            nonce: Some(0),
+            ..Default::default()
+        };
```

Comment:
I suggest we move this temp transaction to the evm sdk and get it from there for both reth and evm canister. And also I'd suggest to use the block number as a parameter (for example as `nonce` or `value`), just in case.